zjit/src/hir_type/mod.rs

```diff
@@ -288,7 +289,7 @@ impl Type {
     /// `Type::from_cint(types::CUInt16, 12)`.
     pub fn from_cint(ty: Type, val: i64) -> Type {
```

Comment:
We should probably have a `from_ptr` instead of `from_cint` and not try and smuggle pointers in that way

Reply:
https://github.com/ruby/ruby/pull/14643/commits/81366f0fa6cbb0b404ccb6c7a1c31355aa5732ce